web/ui/mantine-ui/src/state/settingsSlice.ts

```diff
@@ -30,6 +30,32 @@ export const localStorageKeyEnableSyntaxHighlighting =
 export const localStorageKeyEnableLinter = "settings.enableLinter";
 export const localStorageKeyShowAnnotations = "settings.showAnnotations";
 
+// This dynamically/generically determines the pathPrefix by stripping the first known
+// endpoint suffix from the window location path. It works out of the box for both direct
+// hosting and reverse proxy deployments with no additional configurations required.
+const getPathPrefix = (path: string) => {
+  if (path.endsWith("/")) {
+    path = path.slice(0, -1);
+  }
+
+  const pagePaths = [
```

Comment:
why not putting this array as a global const to avoid having to declare it everytime this function is called ?

Reply:
Was considering that before, but then I figured that this function is only called once anyways, so I moved it into the function to clarify that it's not used from anywhere else.